### ISSUE TITLE
"Expecting function" error under Microsoft Edge with zone.js 0.5.2 & Angular 2 alpha 33

### ISSUE THREAD

**Issue by dsebastien** [angular/zone.js#150]
Currently, the 'patchEventTargetMethods' method in utils.js seems to cause an error under Microsoft Edge. The bind operation fails because the function object that is passed is not of type 'function' as expected but is an object instance of type 'FunctionWrapper'.

Here's the output I get in the console:
```
SCRIPT5022: Expecting function got: [object FunctionWrapper]
core.js (44,7)
```

Putting a breakpoint in the bind function, I see that typeof fn returns "object".

I'm not sure whether the issue is actually caused by zone.js, angular or another dependency and I don't really know how to go about and debug it..

Here's a basic plunker that has the issue: http://embed.plnkr.co/GohfcaiNrwVRn8Hq42tM/preview

**Comment by matthewjh** [comment]
I've seen this on IE11 as well. My conclusion was that it only occurs when the developer tools are open and is (as far as I could tell) innocuous. 

Also hard to see what could b e done to fix this. I for one could find zero documentation (or even a mention) of `FunctionWrapper` on the internet.

**Comment by matthewjh** [comment]
Furthermore, I would say that Angular 2 is completely unrelated as I was not using Angular 2 when I encountered this.

**Comment by btford** [comment]
Thanks for the report, @dsebastien and @matthewjh. I chatted with @vicb, and he said he's is going to take a look at this next week.

**Comment by dsebastien** [comment]
FYI FunctionWrapper seems to be part of Microsoft's JScript, their implementation of EcmaScript included in the .Net framework & used in IE. I assume that they're reusing it for MS Edge.

Here's the only doc about it: https://msdn.microsoft.com/en-US/en_us/library/windows/desktop/microsoft.jscript.functionwrapper(v=vs.110).aspx

**Comment by dsebastien** [comment]
Maybe @csuwildcat could be interested, I saw that he wanted to know about library compat issues with MS Edge (cfr https://github.com/angular/angular/issues/3155)

**Comment by bestander** [comment]
subscribed to the issue, yep all Angular 2 apps can't run in Edge.

**Comment by vicb** [comment]
@mlaval you probably have a much deeper experience debugging win issues than I have. Could you please help here ?

It is obviously linked to https://github.com/angular/zone.js/commit/ee4262a7437820480fd6d01131fb5f705a634dda but reverting the commit (ie the function check) is not enough to get the plunker to work.

Probably we should first make sure that zone.js works standalone (ie not integrated into an angular app) as a first step.

**Comment by csuwildcat** [comment]
@dsebastien hey there, how can I help? Is the error narrowed to something specific I can take a look at?

**Comment by zorend** [comment]
for some reason if you go to BrowserDomAdapter

    clearNodes: function(el) {
            while (el.firstChild) {
                el.firstChild.remove();
          },
and replace it with:

    clearNodes: function(el) {
            while (el.firstChild) {
              try{
                el.firstChild.remove();
              }catch(e){
                el.firstChild.parentNode.removeChild(el.firstChild);
              }
            }
          },
the error disappears but nothing shows in the application,
BTW that was causing an error on IE saying that .remove doesn't exist.

**Comment by pkozlowski-opensource** [comment]
@zorend what is the version of ng that you are talking about here? The use of `remove()` was fixed in master already: https://github.com/angular/angular/blob/218b037d987b2b85a3fb567d71424c9819ac8295/modules/angular2/src/dom/browser_adapter.ts#L134

**Comment by zorend** [comment]
@pkozlowski-opensource  you're right, I'm using alpha33, I'll upgrade and see.

**Comment by vicb** [comment]
@csuwildcat any help would be appreciated here. There is no need to be in the context of angular to trigger the error, a simple way to reproduce is:
- clone the repo,
- serve it over http,
- visit `example/basic.html` and open the console, you'll see the error

**Comment by mlaval** [comment]
My findings about this issue:

1) Zone.js works fine in Edge, all unit tests are passing.
2) About the `Expecting function got: [object FunctionWrapper]`, it is not linked to Zone nor Angular 2, but to the dev tools themselves, in both IE11 and Edge. I sent https://github.com/angular/zone.js/pull/155 to work around that. Note that it doesn't prevent an Angular 2 application to work.
3) Angular 2 is broken in Edge for another reason, see https://github.com/angular/angular/pull/3667

**Comment by vicb** [comment]
fixed by #155

**Comment by dkosasih** [comment]
Happen to me when creating file 
`new File([blob], 'fileEdit')`

Then I use this instead:
`<File>blob`
